cmd/ps-tag-onboarding/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"context"
+	"errors"
+	_ "github.com/viniciusgferreira/ps-tag-onboarding-go/docs"
+	"github.com/viniciusgferreira/ps-tag-onboarding-go/internal/adapters/config"
+	"github.com/viniciusgferreira/ps-tag-onboarding-go/internal/adapters/handler/httpserver"
+	"github.com/viniciusgferreira/ps-tag-onboarding-go/internal/adapters/mongodb"
+	"github.com/viniciusgferreira/ps-tag-onboarding-go/internal/adapters/repository"
+	"github.com/viniciusgferreira/ps-tag-onboarding-go/internal/core/service"
+	"log/slog"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+// @title           Tag Onboarding Go API
+// @version         1.0
+// @description     This is an api to manager users.
+
+// @host      localhost:8080
+func main() {
+	cfg := config.New()
+	slog.Info("Starting the application", "app", cfg.App.Name, "env", cfg.App.Env)
+	db, err := mongodb.Connect(*cfg.DB)
+	if err != nil {
+		panic(err)
+	}
+	userRepo := repository.NewUserRepo(db)
+	userService := service.NewUserService(userRepo)
+
+	var serverHandlers []httpserver.HttpHandlers
+	serverHandlers = append(serverHandlers, httpserver.NewUserHandler(*userService))
+	router := httpserver.NewRouter(cfg.HTTP.GinMode, serverHandlers)
+	server := httpserver.NewServer(cfg.HTTP, router)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
```

Comment:
There is no need of a wait group. The application will stays in line 55 until a signal arrives